test/ret_web/controllers/api-internal/change_login_email_controller_test.exs

```diff
@@ -0,0 +1,74 @@
+defmodule RetWeb.ApiInternal.V1.ChangeLoginEmailControllerTest do
+  use RetWeb.ConnCase
+  import Ret.TestHelpers
+  alias Ret.Account
+
+  @dashboard_access_header "x-ret-dashboard-access-key"
+  @dashboard_access_key "test-key"
+
+  setup_all do
+    merge_module_config(:ret, RetWeb.Plugs.DashboardHeaderAuthorization, %{dashboard_access_key: @dashboard_access_key})
+
+    on_exit(fn ->
+      Ret.TestHelpers.merge_module_config(:ret, RetWeb.Plugs.DashboardHeaderAuthorization, %{dashboard_access_key: nil})
+    end)
+  end
+
+  test "new email addresses must be valid", %{conn: conn} do
+    Account.find_or_create_account_for_email("alice@reticulum.io")
+    assert Account.exists_for_email?("alice@reticulum.io")
+    assert %{status: 400} = post_change_email_for_login(conn, "not_an_email_address", "alice@reticulum.io")
+    refute Account.exists_for_email?("not_an_email_address")
+    assert Account.exists_for_email?("alice@reticulum.io")
+  end
+
+  test "email addresses validation only applies to new emails", %{conn: conn} do
+    Account.find_or_create_account_for_email("not_an_email_address")
+    assert Account.exists_for_email?("not_an_email_address")
+    assert %{status: 200} = post_change_email_for_login(conn, "alice@reticulum.io", "not_an_email_address")
+    assert Account.exists_for_email?("alice@reticulum.io")
+    refute Account.exists_for_email?("not_an_email_address")
+  end
+
+  test "account emails can be changed", %{conn: conn} do
+    refute Account.exists_for_email?("alice@reticulum.io")
```

Comment:
Don’t verify the preconditions of your setup, only the preconditions of the function under test. Cut this line.